src/webhook/mutation/pod_mutator.go

```diff
@@ -725,18 +733,16 @@ func updateContainerOneAgent(c *corev1.Container, dk *dynatracev1beta1.DynaKube,
 		})
 
 	if dk.Spec.Proxy != nil && (dk.Spec.Proxy.Value != "" || dk.Spec.Proxy.ValueFrom != "") {
-		c.Env = append(c.Env,
-			corev1.EnvVar{
-				Name: "DT_PROXY",
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: dtwebhook.SecretConfigName,
-						},
-						Key: "proxy",
-					},
-				},
-			})
+		secretConfig, err := m.getSecretConfigData(nsName)
```

Comment:
Why does the data have to be fetched and passed to the env var instead of setting the secret as the source like before?

Reply:
@meik99 The previous code was assuming that the `proxy` field was an actual field name within the secret, but it was actually embedded within a serialized json string value of the `data.config` field. This was preventing the monitored pod from starting, as k8s would fail it with `CreateContainerError` as the field `config.proxy` does not exist within the secret.

I guess you can't use k8s downward API to reference field names nested within serialized/string values of another field. Thus, I had to fetch the secret and unmarshal the json string value of the `config` field so that I could reference the value of the `proxy` field contained within it.

If there's a better or more elegant way of doing this, I'm open to all suggestions/critique.

Reply:
@meik99 Also, because I needed a client, I had to add the receiver to the `*podMutator` all the way up the call chain.